net-libs/libwebsockets/libwebsockets-3.0.0.recipe

```diff
@@ -0,0 +1,81 @@
+SUMMARY="C library for lightweight websocket clients and servers"
+DESCRIPTION="Libwebsockets (LWS) is a flexible, lightweight pure C library \
+for implementing modern network protocols easily with a tiny footprint, \
+using a nonblocking event loop."
+HOMEPAGE="https://libwebsockets.org/"
+COPYRIGHT="2010-2018 Andy Green"
+LICENSE="GNU LGPL v2.1"
+REVISION="1"
+SOURCE_URI="https://github.com/warmcat/libwebsockets/archive/v$portVersion.tar.gz"
+CHECKSUM_SHA256="a6b611c212c52f161f70556339fdaa199b7e9b6a167c4638e086d19db75d6290"
+SOURCE_DIR="libwebsockets-$portVersion"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	libwebsockets$secondaryArchSuffix = $portVersion
+	lib:libwebsockets$secondaryArchSuffix = $portVersion
+	"
+
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libcrypto$secondaryArchSuffix
+	lib:libssl$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	libwebsockets${secondaryArchSuffix}_devel = $portVersion
+	devel:libwebsockets$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	libwebsockets$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libcrypto$secondaryArchSuffix
+	devel:libssl$secondaryArchSuffix
+	devel:libz$secondaryArchSuffix
+	"
+
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:cmake
+	cmd:make
+	cmd:openssl
+	"
+
+BUILD()
+{
+	mkdir -p build && cd build
+	
+	cmake \
+		-DCMAKE_INSTALL_PREFIX=$prefix \
+		-DCMAKE_BUILD_TYPE=release \
+		-DLWS_INSTALL_INCLUDE_DIR=$includeDir \
+		-DLWS_INSTALL_LIB_DIR=$libDir \
+		-DLWS_WITHOUT_EXTENSIONS=OFF \
+		-DLWS_WITHOUT_TESTAPPS=ON \
+		-DLWS_WITH_HTTP2=ON \
+		-DLWS_WITH_CGI=ON \
+		-DLWS_IPV6=ON \
+		-DLWS_WITH_ZIP_FOPS=ON \
+		-DLWS_WITH_RANGES=ON \
+		-DLWS_WITH_ACME=ON \
+		-DLWS_WITH_SERVER_STATUS=ON \
+		..
+	make $jobArgs
+}
+
+INSTALL()
+{
+	cd build
+	make install
+	
+	prepareInstalledDevelLibs libwebsockets
+	fixPkgconfig
+	
+	packageEntries devel $developDir
```

Comment:
```diff
-	packageEntries devel $developDir
+	packageEntries devel \
+		"$developDir" \
+		"$libDir"/cmake \
+
```
BTW, for some unknown reason, I'm getting the pc files in the base package, instead of the devel package:
```bash
package list -p libwebsockets_x86-3.0.0-1-x86_gcc2.hpkg | grep pkgconfig
lib/pkgconfig
lib/pkgconfig/libwebsockets.pc
lib/pkgconfig/libwebsockets_static.pc
```
(I did not figure out yet how to fix this.)
On x86 secondary arch, all I get for the devel package is:
```bash
package list -p libwebsockets_x86_devel-3.0.0-1-x86_gcc2.hpkg 
develop
develop/headers
develop/headers/x86
develop/headers/x86/libwebsockets.h
develop/headers/x86/lws-plugin-ssh.h
develop/headers/x86/lws_config.h
develop/lib
develop/lib/x86
develop/lib/x86/libwebsockets.a
develop/lib/x86/libwebsockets.so
develop/lib/x86/libwebsockets.so.13
.PackageInfo
```
We should normaly also have in the devel package these files:
```
develop/lib/x86/pkgconfig/libwebsockets.pc
develop/lib/x86/pkgconfig/libwebsockets_static.pc
```